### PR TITLE
[@types/critters-webpack-plugin] Update critters-webpack-plugin to 3.0.2

### PR DESCRIPTION
**Pull request by Gehbt**
Update type to support plugin type for webpack v5.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/critters-webpack-plugin/critters-webpack-plugin-tests.ts
1. types/critters-webpack-plugin/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ac1197a95d15a7b24b1b9d2f6ada0b09287839a1
</p>

File: types/critters-webpack-plugin/index.d.ts [hunk 1]
```diff
@@ -1,8 +1,13 @@
-import { Plugin } from "webpack";
+import { Compiler, WebpackPluginInstance } from "webpack";
 
 export default Critters;
-declare class Critters extends Plugin {
+declare class Critters implements WebpackPluginInstance {
     constructor(options?: Critters.CrittersOptions);
+    apply: (compiler: Compiler) => void;
+    getCssAsset(href: any, style: any): Promise<any>;
```
Comment (JuanJoseGonGi) [inline types/critters-webpack-plugin/index.d.ts:7]: Based on https://github.com/GoogleChromeLabs/critters/blob/main/packages/critters/src/index.js#L302

This method only accepts one argument of type `string`

Comment (JuanJoseGonGi) [inline types/critters-webpack-plugin/index.d.ts:7]: Based on [definition](https://github.com/GoogleChromeLabs/critters/blob/05f5a48d8bcec091cda37fa4fc97a895b33db833/packages/critters-webpack-plugin/src/index.js#L146) first argument is an string and second one an HTMLStyleElement

**Review by JuanJoseGonGi — APPROVED** [review]

**Comment by Gehbt** [comment]
Ready to merge

**Pull request merged**